sql: flake in TestTrace/Session

On [this](https://teamcity.cockroachdb.com/viewLog.html?buildId=9291838&buildTypeId=Cockroach_BazelEssentialCi) CI run (with seemingly unrelated PRs)
```
Failed
=== RUN   TestTrace/Session/TracingOn/node-2
    trace_test.go:391: expected span: "optimizer", got: "local proposal"
    trace_test.go:402: remaining span: "optimizer"
    trace_test.go:402: remaining span: "session recording"
    trace_test.go:402: remaining span: "sql query"
    trace_test.go:402: remaining span: "sql txn"
    trace_test.go:402: remaining span: "table reader"
    trace_test.go:402: remaining span: "txn coordinator send"
            --- FAIL: TestTrace/Session/TracingOn/node-2 (0.04s)
```

Jira issue: CRDB-26102

--- a/pkg/sql/trace_test.go
+++ b/pkg/sql/trace_test.go
@@ -50,6 +50,7 @@ func TestTrace(t *testing.T) {
 		"outbox",
 		"request range lease",
 		"range lookup",
+		"local proposal",
 	}
 
 	testData := []struct {

